core/src/test/java/org/apache/accumulo/core/fate/zookeeper/ServiceLockTest.java

```diff
@@ -66,15 +68,39 @@ public void testSortAndFindLowestPrevPrefix() throws Exception {
         ServiceLock.findLowestPrevPrefix(validChildren,
             "zlock#00000000-0000-0000-0000-eeeeeeeeeeee#0000000010"));
 
-    assertThrows(IndexOutOfBoundsException.class, () -> {
-      ServiceLock.findLowestPrevPrefix(validChildren,
-          "zlock#00000000-0000-0000-0000-aaaaaaaaaaaa#0000000001");
-    });
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> ServiceLock.findLowestPrevPrefix(validChildren,
+            "zlock#00000000-0000-0000-0000-aaaaaaaaaaaa#0000000001"));
 
-    assertThrows(IndexOutOfBoundsException.class, () -> {
-      ServiceLock.findLowestPrevPrefix(validChildren,
-          "zlock#00000000-0000-0000-0000-XXXXXXXXXXXX#0000000099");
-    });
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> ServiceLock.findLowestPrevPrefix(validChildren,
+            "zlock#00000000-0000-0000-0000-XXXXXXXXXXXX#0000000099"));
   }
 
+  @Test
+  public void rejectInvalidUUID() {
+    List<String> children = new ArrayList<>();
+    String uuid = "1-1-1-1-1";
+    String seq = "1234567891";
+    children.add("zlock#" + uuid + "#" + seq);
+
+    // pass as UUID, but fail on string compare.
+    assertEquals("00000001-0001-0001-0001-000000000001", UUID.fromString(uuid).toString());
+    final List<String> validChildren = ServiceLock.validateAndSort(ServiceLock.path(""), children);
+    assertEquals(0, validChildren.size());
+  }
+
+  @Test
+  public void uuidTest() {
+    List<String> children = new ArrayList<>();
+    String uuid = "219ad0f6-ebe0-416e-a20f-c0f32922841d";
+    String seq = "1234567891";
+    children.add("zlock#" + uuid + "#" + seq);
+
+    final List<String> validChildren = ServiceLock.validateAndSort(ServiceLock.path(""), children);
+    assertEquals(1, validChildren.size());
+    String candidate = validChildren.get(0);
+    assertTrue(candidate.contains(uuid));
+    assertTrue(candidate.contains(seq));
```

Comment:
Would it be better to validate the entire string vs validating possible sections? 

```suggestion
    String candidate =  "zlock#" + uuid + "#" + seq;
    children.add(candidate);

    final List<String> validChildren = ServiceLock.validateAndSort(ServiceLock.path(""), children);
    assertEquals(1, validChildren.size());
    assertEquals(candidate, validChildren.get(0));
```

Reply:
The code in `ServiceLock.validateAndSort` parses and tests portions, but if the validation passes, it saves / returns the original strings.  Testing either way then is not really a strong test.  I was using this test to figure out where things were going wrong with the validation. I kept it because it was easier to use this to debug.  The test above does test that the candidate string is returned correctly.  If this test is not adding much value it can be deleted.